### PR TITLE
Prevent processing media if the Uri is null or if mLocalMediaUris is …

### PR DESCRIPTION
**Pull request by geriux**
Fixes https://github.com/wordpress-mobile/WordPress-Android/issues/19943
Fixes https://github.com/wordpress-mobile/WordPress-Android/issues/19944

This is a regression from https://github.com/wordpress-mobile/WordPress-Android/pull/19754 

It will prevent the app from crashing when `mLocalMediaUris` is empty.

There are a few cases where the media image can't be fetched and this prevents the app from crashing, the image won't be added but at least the app remains open. This is a [known issue](https://github.com/wordpress-mobile/WordPress-Android/issues/19373).

-----

## To Test:

> [!NOTE]  
> I could only reproduce it on a Redmi Note 8T using Android 11 while sharing an image from the File Explorer app. This prevents a crash when the media can't be added, now the app will remain open but the Image won't be added, hopefully will work on fixing [this issue](https://github.com/wordpress-mobile/WordPress-Android/issues/19373) soon. So I propose testing the sharing feature overall:

- **Precondition**: Have different kinds of files on your device: `PDF`, `jpg`,  and `mp4`.
- **NOTE:** The editor takes a bit of time to load when adding media. [GitHub ticket](https://github.com/wordpress-mobile/gutenberg-mobile/issues/6529).
- Install the app from this PR
- Open your Files app
- Multiselect an image and a PDF file 
- Tap on the Share button
- Select the Jetpack app
- **Expect** the editor to be loaded with just the supported media types (images, and videos)

### Extra test case
- Open your Files app
- Select one image
- Select the Jetpack app
- **Expect** the editor to be loaded with the selected image

Test Case 1|Test Case 2
-|-
<video src="https://github.com/WordPress/gutenberg/assets/4885740/dd8dd610-5371-4da0-acf2-a36d1914c97e" />|<video src="https://github.com/wordpress-mobile/WordPress-Android/assets/4885740/19574a54-0db9-41bf-ab45-e30e8cb80278" />

-----

## Regression Notes

1. Potential unintended areas of impact

    - Sharing Media

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

3. What automated tests I added (or what prevented me from doing so)

N/A
-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/ShareIntentReceiverActivity.java [hunk 1]
```diff
@@ -110,8 +110,11 @@ private void downloadExternalMedia() {
     }
 
     private boolean isAllowedMediaType(@NonNull Uri uri) {
```
Comment (antonis) [inline WordPress/src/main/java/org/wordpress/android/ui/ShareIntentReceiverActivity.java]: Great work @geriux 🙇 
Given that `isAllowedMediaType` has a `NonNull` annotation for the `uri` parameter I would suggest moving the null check on the two call sites instead (if statements on line 100 and 106 above). Wdyt?

**Review by antonis — COMMENTED** [review]
Thank you for the quick fix @geriux 🙇 
I wasn't able to reproduce the issue but checking the crash stack traces your fixes look good to me 👍 
I left a suggestion at https://github.com/wordpress-mobile/WordPress-Android/pull/19947/files#r1452499476

File: WordPress/src/main/java/org/wordpress/android/ui/ShareIntentReceiverActivity.java (hunk 1 above)
Comment (geriux) [reply WordPress/src/main/java/org/wordpress/android/ui/ShareIntentReceiverActivity.java]: Thanks for the suggestion! I'll update it 🚀

Comment (geriux) [reply WordPress/src/main/java/org/wordpress/android/ui/ShareIntentReceiverActivity.java]: Updated in https://github.com/wordpress-mobile/WordPress-Android/pull/19947/commits/e837837053d114602301ec68fec1907bd90579a8

**Comment by geriux** [comment]
> I wasn't able to reproduce the issue but checking the crash stack traces your fixes look good to me 👍

Yeah I only was able to reproduce it on one of my testing devices, here's a recording of the steps I followed to get it to crash:

https://github.com/wordpress-mobile/WordPress-Android/assets/4885740/268c093d-88a1-412f-bcb4-e2a0a6f5d7aa

File: WordPress/src/main/java/org/wordpress/android/ui/ShareIntentReceiverActivity.java (hunk 1 above)
Comment (antonis) [reply WordPress/src/main/java/org/wordpress/android/ui/ShareIntentReceiverActivity.java]: Thank you @geriux 🙇

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-Android/milestone/273">24.0 ❄️</a>. The due date for this milestone has already passed.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by antonis** [comment]
> Yeah I only was able to reproduce it on one of my testing devices, here's a recording of the steps I followed to get it to crash

Thank you for the recording @geriux 🙇 
I was able to trigger the crash on debug mode an verified the fix :)

**Review by antonis — APPROVED** [review]
Awesome work @geriux!
Thank you for fixing this so quickly. I verified that the code changes fix both https://github.com/wordpress-mobile/WordPress-Android/issues/19943 and https://github.com/wordpress-mobile/WordPress-Android/issues/19944 🎉

**Pull request merged**